qstarz_bl_1000.cc

```diff
@@ -0,0 +1,297 @@
+  stream >> speed;
+  stream >> altitude;
+  stream >> heading;
+
+  stream >> gx;
+  stream >> gy;
+  stream >> gz;
+  stream >> maxSNR;
+
+  stream.setFloatingPointPrecision(QDataStream::SinglePrecision);
+  stream >> hdop;
+  stream >> vdop;
+
+  stream >> satelliteCountView;
+  stream >> satelliteCountUsed;
+  stream >> fixQuality;
+  stream >> batteryPercent;
+  stream >> unused1;
+  stream >> unused2;
+
+  if (stream.status() != QDataStream::Ok) {
+    Fatal() << MYNAME << ": File format error on " << read_fname << ". Perhaps this isn't a Qstarz BL-1000 file";
+  }
+
+  BL1000_POINT_TYPE type;
+
+  switch (rcr) {
+  case 'B':
+    type = BL1000_POINT_TYPE_WAY_POINT;
+    break;
+  case 'T':
+    type = BL1000_POINT_TYPE_TRACK_POINT_BY_TIME;
+    break;
+  case 'D':
+    type = BL1000_POINT_TYPE_TRACK_POINT_BY_DISTANCE;
+    break;
+  case 'S':
+    type = BL1000_POINT_TYPE_TRACK_POINT_BY_SPEED;
+    break;
+  case 'W': // Power On
+  case 'X': // Power Off
+  case 'Y': // USB plug in
+  case 'Z': // USB removed
+  case 'C': // Cali G-sensor
+  case 'A': // Enter Smart Mode(Power Saving)
+  case 'R': // Reserved
+  case 'N': // Reserved
+  case 'M': // Reserved
+  case 'G': // Reserved
+  case 'H': // Reserved
+  case 'I': // Reserved
+  case 'J': // Reserved
+    type = BL1000_POINT_TYPE_UNKNOWN;
+    break;
+  default:
+    type = BL1000_POINT_TYPE_UNKNOWN;
+
+    Fatal() << MYNAME << ": File format error on " << read_fname << ". Unexpected value for RCR (record reason): " << rcr;
+
+    break;
+  }
+
+  fix_type fix;
+
+  switch (fixQuality) {
+  case 1:
+    switch (fixStatus) {
+    case 1:
+      fix = fix_none;
+      break;
+    case 2:
+      fix = fix_2d;
+      break;
+    case 3:
+      fix = fix_3d;
+      break;
+    default:
+      fix = fix_unknown;
+      break;
+    }
+    break;
+  case 2:
+    fix = fix_dgps;
+    break;
+  case 3:
+    fix = fix_pps;
+    break;
+  default:
+    fix = fix_unknown;
+
+    if (type != BL1000_POINT_TYPE_UNKNOWN) {
+      Fatal() << MYNAME << ": File format error on " << read_fname << ". Unexpected value for fix quality: " << fixQuality;
+    }
+
+    break;
+  }
+
+  auto* waypoint = new Waypoint();
+
+  waypoint->latitude = ddmm2degrees(dLatitude);
+  waypoint->longitude = ddmm2degrees(dLongitude);
+
+  // qDebug(waypoint)
+
+  if ((waypoint->latitude < -90) || (waypoint->latitude > 90)) {
+    Fatal() << MYNAME << ": File format error on " << read_fname << ". Unexpected value for latitude: " << waypoint->latitude;
+  }
+
+  if ((waypoint->longitude < -180) || (waypoint->longitude > 180)) {
+    Fatal() << MYNAME << ": File format error on " << read_fname << ". Unexpected value for longitude: " << waypoint->longitude;
+  }
+
+  waypoint->altitude = altitude;
+
+  waypoint->hdop = round(hdop * 1000) / 1000;
+  waypoint->vdop = round(vdop * 1000) / 1000;
+  waypoint->fix = fix;
+  waypoint->sat = satelliteCountUsed;
+
+  waypoint->speed = speed * (1000.0 / (60 * 60)); // km/h to m/s
```

Comment:
I am glad we figured we needed a unit conversion here.  We have this conversion defined in defs.h.

> /* kilometers/hour to meters/second */
constexpr double KPH_TO_MPS(double a) { return a * 1000.0/SECONDS_PER_HOUR;}

So you could say
waypoint->speed = KPH_TO_MPS(speed);